src/py21cmfast/wrapper.py

```diff
@@ -2712,22 +2711,31 @@ def run_lightcone(
                 regenerate=regenerate,
             )
 
-            if coeval_callback is not None:
-                if compute_coeval_callback[iz]:
-                    coeval = Coeval(
-                        redshift=z,
-                        initial_conditions=init_box,
-                        perturbed_field=pf2,
-                        ionized_box=ib2,
-                        brightness_temp=bt2,
-                        ts_box=st2 if flag_options.USE_TS_FLUCT else None,
-                        photon_nonconservation_data=_get_photon_nonconservation_data()
-                        if flag_options.PHOTON_CONS
-                        else None,
-                        _globals=None,
-                    )
+            if coeval_callback is not None and compute_coeval_callback[iz]:
+                coeval = Coeval(
+                    redshift=z,
+                    initial_conditions=init_box,
+                    perturbed_field=pf2,
+                    ionized_box=ib2,
+                    brightness_temp=bt2,
+                    ts_box=st2 if flag_options.USE_TS_FLUCT else None,
+                    photon_nonconservation_data=_get_photon_nonconservation_data()
+                    if flag_options.PHOTON_CONS
+                    else None,
+                    _globals=None,
+                )
+                try:
                     coeval_callback_output.append(coeval_callback(coeval))
-
+                except Exception as e:
+                    if sum(compute_coeval_callback[: iz + 1]) == 1:
+                        raise RuntimeError(
+                            f"coeval_callback computation failed on first trial, z={z}."
+                        )
+                    else:
+                        logger.warning(
+                            f"coeval_callback computation failed on z={z}, skipping."
+                        )
+                        logger.warning(e)
```

Comment:
I *think* this won't work -- you might need `e.message` or something like that. You should be able to test this to make sure (as a pytest test, use the `pytest.warns` function to check that a suitable warning is raised).

Reply:
Would instead of the two the following work: `logger.exception(f"coeval_callback bla bla")` ?

Reply:
I'm not sure exactly what you mean? Even to the `logger.exception` I think you'll have to give it a string, not an actual `Exception` instance. I can't remember if calling `logger.exception` actually raises an exception or not...

Reply:
Okay I checked this, exception class has default `__str__` method, so both `print(e)` or `logger.warning(e)` or similar will work. However this doesn't print exact exception class name but only final error message, so I think something like `f"{type(e).__name__}, {e}"` should be sufficient. `logger.exception("message")` prints out "message" and full traceback, which is maybe redundant?